scripts/irods/test/test_iquest.py

```diff
@@ -148,3 +149,136 @@ def test_iquest_bad_format(self):
     def test_iquest_incorrect_format_count(self):
         self.admin.assert_icommand("iquest \"%s %s\" \"select COLL_NAME where COLL_NAME like '%home%'\"",
                                    'STDERR_SINGLELINE', 'boost::too_few_args: format-string referred to more arguments than were passed')
+
+
+class test_iquest_with_data_resc_hier(unittest.TestCase):
+    @classmethod
+    def setUpClass(self):
+        self.user = session.mkuser_and_return_session('rodsuser', 'alice', 'apass', lib.get_hostname())
+
+        self.root_resource = 'root'
+        self.mid_resource = 'mid'
+        self.leaf_resource = 'leaf'
+        self.data_name = 'foo'
+        self.logical_path = os.path.join(self.user.session_collection, self.data_name)
+        self.hierarchy = ';'.join([self.root_resource, self.mid_resource, self.leaf_resource])
+
+        with session.make_session_for_existing_admin() as admin_session:
+            lib.create_passthru_resource(self.root_resource, admin_session)
+            lib.create_passthru_resource(self.mid_resource, admin_session)
+            lib.create_ufs_resource(self.leaf_resource, admin_session)
+            lib.add_child_resource(self.root_resource, self.mid_resource, admin_session)
+            lib.add_child_resource(self.mid_resource, self.leaf_resource, admin_session)
+
+        self.user.assert_icommand(['itouch', '-R', self.leaf_resource, self.logical_path])
+
+
+    @classmethod
+    def tearDownClass(self):
+        self.user.assert_icommand(['irm', '-f', self.logical_path])
+
+        with session.make_session_for_existing_admin() as admin_session:
+            self.user.__exit__()
```

Comment:
Right, this class is not using the base class that a lot of other tests use. This is required to clean up the session for the user created on 157.

I went this route just because I wanted to keep the test class trim-ish, but I'm happy to use the existing utilities if that seems better

Reply:
No, it's fine as is. I was just curious.